### PR TITLE
Quests now support multiple channels

### PR DESCRIPTION
**Pull request by zen0-5338**
Quests can now be made to spawn in specific channels instead of only one

### CODE REVIEW

File: dueutil/botcommands/quest.py [hunk 1]
```diff
@@ -435,12 +435,23 @@ async def editquest(ctx, quest_name, updates, **_):
...
+                value = [id for l in value for id in l] # flatten the list - final output will always be list with list containing single elements
+                quest_channels = []
+                channel_list_string = ""
+                for channel in value:
+                    channel_id = channel.replace("<#", "").replace(">", "")
+                    ret_channel = util.get_channel(channel_id)
+                    if ret_channel is not None:
+                        if ret_channel not in quest_channels: # avoid duplicates
+                            quest_channels.append(channel_id)
+                            channel_list_string += f'\n\t{ret_channel.mention}'
```
Comment (DeveloperAnonymous) [inline dueutil/botcommands/quest.py:449]: Ew, use double quotes

File: dueutil/botcommands/quest.py [hunk 2]
```diff
@@ -435,12 +435,23 @@ async def editquest(ctx, quest_name, updates, **_):
...
             else:
-                channel_id = value.replace("<#", "").replace(">", "")
-                channel = util.get_channel(channel_id)
-                if channel is not None:
-                    quest.channel = channel.id
-                else:
-                    updates[quest_property] = "Channel not found!"
+                value = [i.strip() for i in value.split(',')] # split through comma
+                value = [i.split() for i in value] #split through space
+                value = [id for l in value for id in l] # flatten the list - final output will always be list with list containing single elements
```
Comment (DeveloperAnonymous) [inline dueutil/botcommands/quest.py:440]: tbh I'm really not a big fan of overriding a variable multiple times and especially one that could potentially be used else where

File: dueutil/game/quests.py [hunk 3]
```diff
@@ -94,14 +94,17 @@ def base_values(self):
                                self.base_accy, self.base_hp, )
 
     def get_channel_mention(self, guild):
+        ret = ''
```
Comment (DeveloperAnonymous) [inline dueutil/game/quests.py:97]: Ew

File: dueutil/game/quests.py [hunk 4]
```diff
@@ -94,14 +94,17 @@ def base_values(self):
...
         else:
-            channel = guild.get_channel(self.channel)
-            if channel is None:
-                return "``Deleted``"
-            else:
-                return channel.mention
+            for channel in self.channel:
+                channel = guild.get_channel(int(channel))
+                if channel is None:
+                    ret+="\n``Deleted``"
```
Comment (DeveloperAnonymous) [inline dueutil/game/quests.py:104]: Install black to format.. it should be `ret += "\n``Deleted``"`

File: dueutil/game/quests.py [hunk 5]
```diff
@@ -324,8 +327,8 @@ def load_default_quests():
         loaded_quest: Quest = jsonpickle.decode(quest['data'])
 
         if isinstance(loaded_quest.channel, str) and loaded_quest.channel not in ("ALL", None, "NONE"):
-            loaded_quest.channel = int(loaded_quest.channel)
-        
+            loaded_quest.channel = eval(loaded_quest.channel)
```
Comment (DeveloperAnonymous) [inline dueutil/game/quests.py:330]: I'm a bit shocked to see eval here, wouldn't there be a better way?

**Review by DeveloperAnonymous — CHANGES_REQUESTED** [review]

File: dueutil/botcommands/quest.py (hunk 1 above)
Comment (zen0-5338) [reply dueutil/botcommands/quest.py:449]: Shut up, single is faster, I use colemak-dh

File: dueutil/game/quests.py (hunk 5 above)
Comment (zen0-5338) [reply dueutil/game/quests.py:330]: Unless you are making the database atomic, no

File: dueutil/botcommands/quest.py (hunk 2 above)
Comment (zen0-5338) [reply dueutil/botcommands/quest.py:440]: it is not going to be used again
better to not use more mem

**Review by DeveloperAnonymous — APPROVED** [review]

**Pull request merged**